sled-agent/src/zone_bundle.rs

```diff
@@ -564,45 +973,1240 @@ pub async fn extract_zone_bundle_metadata(
     task.await?
 }
 
-/// Get the path to a zone bundle, if it exists.
-pub async fn get_zone_bundle_path(
+// Find zone bundles in the provided directory, which match the filter function.
+async fn filter_zone_bundles(
+    log: &Logger,
+    directory: &Utf8PathBuf,
+    filter: impl Fn(&ZoneBundleMetadata) -> bool,
+) -> Result<BTreeMap<Utf8PathBuf, ZoneBundleMetadata>, BundleError> {
+    let mut out = BTreeMap::new();
+    debug!(log, "searching directory for zone bundles"; "directory" => %directory);
+    let mut rd = tokio::fs::read_dir(directory).await?;
+    while let Some(entry) = rd.next_entry().await? {
+        let path = entry.path();
+        debug!(log, "checking path as zone bundle"; "path" => path.display());
+        match extract_zone_bundle_metadata(path.clone()).await {
+            Ok(md) => {
+                trace!(log, "extracted zone bundle metadata"; "metadata" => ?md);
```

Comment:
I believe we build `slog` with `release_max_level_debug`, so I don't think there will ever be trace messages. I am personally of the opinion that we should build with trace, and filter at runtime with a `LevelFilter`. We're already doing the level filter to remove debug messages on `main`, since the level filter's value comes from the sled agent's `config.toml`.

If we do the above, then we can still get the trace messages even with the level filter in place, using DTrace. If you compile them out, you can't get them with DTrace. But I think that's a larger discussion, since it might have a big impact. I guess we'd have to measure to know.